graylog2-server/src/main/java/org/graylog/events/notifications/types/HTTPEventNotificationV2.java

```diff
@@ -0,0 +1,247 @@
+                                   final ParameterizedHttpClientProvider parameterizedHttpClientProvider) {
+        super(whitelistService, urlWhitelistNotificationService, encryptedValueService);
+        this.notificationCallbackService = notificationCallbackService;
+        this.objectMapperProvider = objectMapperProvider;
+        this.configurationProvider = configurationProvider;
+        this.parameterizedHttpClientProvider = parameterizedHttpClientProvider;
+        this.templateEngine = templateEngine;
+    }
+
+    /**
+     * Depending on the configuration, either a default HTTP client will be returned or an instance
+     * with {@link org.graylog2.shared.bindings.providers.TcpKeepAliveSocketFactory} configured.
+     */
+    private OkHttpClient selectClient(HTTPEventNotificationConfigV2 notificationConfig) {
+        final boolean withKeepAlive = configurationProvider.get().notificationsKeepAliveProbe();
+        return parameterizedHttpClientProvider.get(withKeepAlive, notificationConfig.skipTLSVerification());
+    }
+
+    @Override
+    public void execute(EventNotificationContext ctx) throws TemporaryEventNotificationException, PermanentEventNotificationException {
+        final HTTPEventNotificationConfigV2 config = (HTTPEventNotificationConfigV2) ctx.notificationConfig();
+        ImmutableList<MessageSummary> backlog = notificationCallbackService.getBacklogForEvent(ctx);
+        final EventNotificationModelData modelData = EventNotificationModelData.of(ctx, backlog);
+        final HttpUrl httpUrl = validateUrl(config.url(), ctx.notificationId(), modelData.eventDefinitionTitle());
+
+        final Request.Builder builder = new Request.Builder();
+        addAuthHeader(builder, config.basicAuth());
+        addApiKey(builder, httpUrl, config.apiKey(), config.apiSecret());
+
+        final String body;
+        try {
+            body = buildRequestBody(modelData, config);
+        } catch (JsonProcessingException processingErr) {
+            throw new PermanentEventNotificationException("Unable to serialize notification", processingErr);
+        } catch (UnsupportedEncodingException encodingErr) {
+            throw new PermanentEventNotificationException("Unable to URL encode notification body", encodingErr);
+        }
+
+        switch (config.httpMethod()) {
+            case GET -> builder.get();
+            case PUT -> builder.put(RequestBody.create(body, getMediaType(config.contentType())));
+            case POST -> builder.post(RequestBody.create(body, getMediaType(config.contentType())));
+        }
+
+        LOG.debug("Requesting HTTP endpoint at <{}> in notification <{}>", config.url(), ctx.notificationId());
+
+        final OkHttpClient httpClient = selectClient(config);
+        try (final Response r = httpClient.newCall(builder.build()).execute()) {
+            if (!r.isSuccessful()) {
+                throw new PermanentEventNotificationException(
```

Comment:
@kingzacko1 what do you think adding some more context to this error message, like the response message and/or error body message? Just from my testing it would've been helpful to debug.

Also wondering if we should add a system notification when there's a failure like we do with some other notifications?: https://github.com/Graylog2/graylog2-server/blob/7b43330f0755aab1e9f1fdf520f3502a4cf6f93b/graylog2-server/src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java#L110-L117

Reply:
Also going to add this in the next commit!

Reply:
@ryan-carroll-graylog so after playing around with this and looking at some error responses, there really isn't ever going to be much of anything in those error response messages. If I'm missing something let me know, but I think this may be about as good as we can do using this Response object.

Reply:
Ok yeah I think this is solid with the system notification. I was thinking about adding `r.message()` but it's basically just the plain language version of the response code so I don't think it's really necessary.